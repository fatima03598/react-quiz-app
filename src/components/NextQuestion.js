import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { render } from 'react-dom';

const Entities = require('html-entities').AllHtmlEntities;
const entities = new Entities();

 class NextQuestion extends Component {
    constructor(){
        super()
        this.state={
            previousQuestionNumber: null
        }
    }

    componentDidMount() {
        this.getParam();
     }
 
     getParam = () => {
         this.setState({
             previousQuestionNumber: parseInt(this.props.match.params.previousQuestionNum),
         });
     }

    userCorrect = (user) => {
        return(this.props.location.state.feedback[user] === 1 ? user + " was right": user + " was wrong")
        }

    render() {
        let questionNumber = this.state.previousQuestionNumber;
        let nextQuestionNumber = questionNumber + 1;
        let nextQNumberString = nextQuestionNumber.toString();
        let currentQuestionScores = this.props.location.state.questionScores;
        let stateNext = this.props.location.state.questionStateNext;
        return (
            <div className="nextquestion">
                <h1>Next Question</h1>
                <div className="user-feedback">{(Object.values(this.props.location.state.questionStateNext.userNames)).map(user => <h3>{this.userCorrect(user)}</h3>)}</div>
                <h3>{entities.decode(this.props.location.state.questionStateNext.questionsArray[this.props.match.params.previousQuestionNum].question)}</h3>
                <p>The correct answer was: {entities.decode(this.props.location.state.questionStateNext.questionsArray[this.props.match.params.previousQuestionNum].correct_answer)}</p>
                {questionNumber == 9 ? <Link to={{ pathname:"/scores", state:{userScores: currentQuestionScores, stateFinal: stateNext }}}><input type="button" value="View Scores!" /></Link> : <Link to={{pathname:`/question/${nextQNumberString}`, state: {qNumber: nextQuestionNumber, questionState: stateNext, previousQuestionScores: currentQuestionScores}}} ><input type="submit"  value="Next Question"/></Link>}
            </div>
               
        )
    }
}

export default NextQuestion;



