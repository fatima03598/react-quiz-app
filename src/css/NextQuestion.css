.next-question-container{
    display: flex;
    flex-direction: column;
    padding: 2em;
    font-size: 1.5em;
    background-color: #DCEDFC;

}

.user-feedback {
    display: flex;
    flex-direction: column;
    /* justify-content: center; */
    align-items: center;
}


.next-question-button-container {
    display: flex;
    justify-content: center;
    
}

.next-question-button {
    border-radius: 1em;
    border: 0;
    /* width: 4em; */
    font-size: 0.5em;
    padding: 1em;
    background-color: #052848;
    color: white;
}