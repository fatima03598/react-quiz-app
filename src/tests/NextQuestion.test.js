import React from 'react'
import { Link } from 'react-router-dom';
import { shallow, mount } from 'enzyme'
import NextQuestion from '../components/NextQuestion'


describe('NextQuestion', () => {

    let dumbieState = {
        previousQuestionScores:{
            R: 0,
            Ro: 0,
            Ror: 0,
            Rori: 0,
            Rorie: 0
        },
        qNumber: 0,
        questionState: {
            category: "9",
            difficulty: "easy",
            playersNumber: "1",
            questionsArray: [{category: "General Knowledge", type: "multiple", difficulty: "easy", question: "In the video-game franchise Kingdom Hearts, the main protagonist, carries a weapon with what shape?", correct_answer: "Key", incorrect_answers: ["Sword", "Pen", "Cellphone"]}],
            totalScore: 0,
            userNames: {0: "Rorie"},
            userScores: {R: 0, Ro: 0, Ror: 0, Rori: 0, Rorie: 0}
        },
        category: "9",
            difficulty: "easy",
            playersNumber: "1",
            questionsArray: [{category: "General Knowledge", type: "multiple", difficulty: "easy", question: "In the video-game franchise Kingdom Hearts, the main protagonist, carries a weapon with what shape?", correct_answer: "Key", incorrect_answers: ["Sword", "Pen", "Cellphone"]}],
            totalScore: 0,
            userNames: {0: "Rorie"},
            userScores: {R: 0, Ro: 0, Ror: 0, Rori: 0, Rorie: 0}
    }
    let dumbieScores = {R: 0, Ro: 0, Ror: 0, Rori: 0, Rorie: 1}
    let dumbieFeedback = {Rorie: 1}


    it('should render main User Feedback div', () => {
        let wrapper = shallow(<NextQuestion 
            match={{
                isExact: true,
                params: {previousQuestionNum: "0"},
                path: "/nextquestion/:previousQuestionNum",
                url: "/nextquestion/0"}} 
            location={{state: {queNumber: 1, questionStateNext: dumbieState, questionScores: dumbieScores, feedback: dumbieFeedback }}}/>);
        expect(wrapper.exists('.user-feedback')).toEqual(true);
    });

    it('should render the Next Question div', () => {
        let wrapper = shallow(<NextQuestion 
            match={{
                isExact: true,
                params: {previousQuestionNum: "0"},
                path: "/nextquestion/:previousQuestionNum",
                url: "/nextquestion/0"}} 
            location={{state: {queNumber: 1, questionStateNext: dumbieState, questionScores: dumbieScores, feedback: dumbieFeedback }}}/>);
        expect(wrapper.exists('.next-question-button-container')).toEqual(true);
    });

    it('should render the Next Question div', () => {
        let wrapper = shallow(<NextQuestion 
            match={{
                isExact: true,
                params: {previousQuestionNum: "0"},
                path: "/nextquestion/:previousQuestionNum",
                url: "/nextquestion/0"}} 
            location={{state: {queNumber: 1, questionStateNext: dumbieState, questionScores: dumbieScores, feedback: dumbieFeedback }}}/>);
        expect(wrapper.exists('.next-question-container')).toEqual(true);
    });

});







//npm test -- --coverage --watchAll=false  
    