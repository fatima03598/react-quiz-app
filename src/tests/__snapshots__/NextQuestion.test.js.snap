// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextQuestion matches the snapshot 1`] = `
<div
  className="next-question-container"
>
  <h3>
    In the video-game franchise Kingdom Hearts, the main protagonist, carries a weapon with what shape?
  </h3>
  <p>
    The correct answer was: 
    Key
  </p>
  <div
    className="user-feedback"
  >
    <h3>
      Rorie was correct âœ“
    </h3>
  </div>
  <div
    className="next-question-button-container"
  >
    <a
      href="/scores"
      onClick={[Function]}
    >
      <input
        className="next-question-button"
        type="button"
        value="View Scores!"
      />
    </a>
  </div>
</div>
`;
